api/services/SiteMembershipCreator.js

```diff
@@ -66,5 +73,6 @@ const revokeSiteMembership = ({ user, site, userId }) =>
       const userToRemove = site.Users.find(u => u.id === Number(userId));
```

Comment:
here's I think where you could add a check to make sure  the owner isn't getting removed:

```js
if (userToRemove.username.toLowerCase() === site.owner.toLowerCase()) {
 throw { status: 400, message: 'The site owner cannot be removed' };
}
```